### PR TITLE
Create target

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello references

* [As a user I should be able to create a new target so I can find people with similar interests in a specific area](https://trello.com/c/ly8TdHic)
* [As a user I should only be able to create a maximum of 10 targets so the map doesn't get too crowded](https://trello.com/c/tZsKsCVT)

#### Description

* Create a new target when the user clicks on the map
* Valid form data
* Create select to show topic values and icon

create

### CODE REVIEW

File: src/components/TargetsMap/index.js [hunk 1]
```diff
@@ -16,18 +20,39 @@ const TargetsMap = () => {
...
+  const history = useHistory();
 
   const dispatch = useDispatch();
   useEffect(() => {
     dispatch(getTargets(1));
     dispatch(getTopics());
   }, []);
 
+  const handleNewTarget = location => {
+    dispatch(addTarget(location));
```
Comment (peter7z) [inline src/components/TargetsMap/index.js:33]: Is just an alternative, but this info could skip redux and be passed directly.

**Review by peter7z — APPROVED** [review]

**Review by peter7z — APPROVED** [review]

File: src/components/TargetsMap/index.js (hunk 1 above)
Comment (afmicc) [reply src/components/TargetsMap/index.js:33]: do you mean query string params?

File: src/components/TargetForm/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  const dispatch = useDispatch();
+  const error = useSelector(({ target: { error } }) => error);
+  const topics = useSelector(({ topic: { values } }) =>
+    values?.map(({ topic: { id, label, icon } }) => ({ key: id, value: label, icon }))
+  );
+
+  const handleFormSubmit = event => {
+    event.preventDefault();
+    if (validTarget) dispatch(saveTarget({ ...data, lat, lng }));
```
Comment (manema) [inline src/components/TargetForm/index.js]:
```suggestion
    validTarget && dispatch(saveTarget({ ...data, lat, lng }));
```

File: src/components/common/Select/index.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { arrayOf, bool, func, number, oneOfType, string, shape } from 'prop-types';
+
+import './Select.scss';
+
+const Select = ({ title, placeholder, options, valid, value, onChange }) => {
+  const [display, setDisplay] = useState(false);
```
Comment (manema) [inline src/components/common/Select/index.js]: You can omit that false and useState will be initialized with a falsy value anyway.

**Review by manema — APPROVED** [review]

**Pull request merged**